distributed/nanny.py

```diff
@@ -498,7 +498,7 @@ def run(self, comm, *args, **kwargs):
         return run(self, comm, *args, **kwargs)
 
     def _on_exit_sync(self, exitcode):
-        self.loop.add_callback(self._on_exit, exitcode)
+        self._ongoing_background_tasks.call_soon(self._on_exit, exitcode)
```

Comment:
Was this method relying on the fact that `loop.add_callback` is threadsafe, and can be called from outside the event loop? The `sync` part of the name makes me think that. I'm not sure if our new `call_soon` is also threadsafe.

Reply:
no `_on_exit_sync` is run in a `loop.add_callback()`


https://github.com/dask/distributed/blob/8873ccbf0d6b0c067b7784902b5d5bd51644db69/distributed/nanny.py#L408
https://github.com/dask/distributed/blob/8873ccbf0d6b0c067b7784902b5d5bd51644db69/distributed/nanny.py#L660
https://github.com/dask/distributed/blob/8873ccbf0d6b0c067b7784902b5d5bd51644db69/distributed/process.py#L314-L326
https://github.com/dask/distributed/blob/8873ccbf0d6b0c067b7784902b5d5bd51644db69/distributed/process.py#L240
https://github.com/dask/distributed/blob/8873ccbf0d6b0c067b7784902b5d5bd51644db69/distributed/process.py#L129-L134